lib/workers/repository/update/branch/index.ts

```diff
@@ -371,34 +374,69 @@ export async function processBranch(
-        ) {
-          const timeElapsed = getElapsedMs(upgrade.releaseTimestamp);
-          if (timeElapsed < coerceNumber(toMs(upgrade.minimumReleaseAge))) {
-            logger.debug(
-              {
-                depName: upgrade.depName,
-                timeElapsed,
-                minimumReleaseAge: upgrade.minimumReleaseAge,
-              },
-              'Update has not passed minimum release age',
-            );
-            config.stabilityStatus = 'yellow';
-            continue;
+        if (is.nonEmptyString(upgrade.minimumReleaseAge)) {
+          let minimumReleaseAgeTimestamp: MinimumReleaseAgeTimestamp =
+            'required';
+          if (
+            isMinimumReleaseAgeTimestamp(upgrade.minimumReleaseAgeTimestamp)
+          ) {
+            minimumReleaseAgeTimestamp = upgrade.minimumReleaseAgeTimestamp;
+          }
+
+          // regardless of the value of `minimumReleaseAgeTimestamp`, if there is a timestamp, we will process it according to `minimumReleaseAge`
+          if (upgrade.releaseTimestamp) {
+            const timeElapsed = getElapsedMs(upgrade.releaseTimestamp);
+            if (timeElapsed < coerceNumber(toMs(upgrade.minimumReleaseAge))) {
+              logger.debug(
+                {
+                  depName: upgrade.depName,
+                  timeElapsed,
+                  minimumReleaseAge: upgrade.minimumReleaseAge,
+                },
+                'Update has not passed minimum release age',
+              );
+              config.stabilityStatus = 'yellow';
+              continue;
+            }
+          } else {
+            // if we're set to `minimumReleaseAgeTimestamp=required`, and there isn't a timestamp, always mark the update as pending
+            if (minimumReleaseAgeTimestamp === 'required') {
+              depNamesWithoutReleaseTimestamp.required.push(upgrade.depName!)
+              config.stabilityStatus = 'yellow';
+              continue;
+            } else {
+              // if there is no timestamp, and we're running in `optional` mode, we can allow it, but make sure to warn the user
+              depNamesWithoutReleaseTimestamp.optional.push(upgrade.depName!)
+            }
           }
         }
+
+        if (depNamesWithoutReleaseTimestamp.required) {
+          logger.debug(
+            { depNames: depNamesWithoutReleaseTimestamp.required, },
+            `${depNamesWithoutReleaseTimestamp.required.length} upgrades did not have a releaseTimestamp, and as we're running with minimumReleaseAgeTimestamp=required, these upgrades(s) will be marked as pending status checks`,
+          );
+        } else if (depNamesWithoutReleaseTimestamp.optional) {
+          logger.warn(
+            { depNames: depNamesWithoutReleaseTimestamp.optional, },
+            `${depNamesWithoutReleaseTimestamp.optional.length} upgrades did not have a releaseTimestamp, but as we're running with minimumReleaseAgeTimestamp=optional, proceeding`,
+          );
+        }
+
```

Comment:
Should we find the unique `depName`s here, or are we happy that we may be logging the same `depName` multiple times?